two-fer/two_fer.go

```diff
@@ -1,15 +1,15 @@
-// This is a "stub" file.  It's a little start on your solution.
-// It's not a complete solution though; you have to write some code.
-
-// Package twofer should have a package comment that summarizes what it's about.
-// https://golang.org/doc/effective_go.html#commentary
 package twofer
 
-// ShareWith should have a comment documenting it.
+import (
+	"fmt"
+)
+
 func ShareWith(name string) string {
-	// Write some code here to pass the test suite.
-	// Then remove all the stock comments.
-	// They're here to help you get started but they only clutter a finished solution.
-	// If you leave them in, reviewers may protest!
-	return ""
-}
+
+	if len(name) > 0 {
+		return fmt.Sprintf("One for %v, one for me.", name)
+	} else {
+		return "One for you, one for me."
+	}
+
```

Comment:
Yep, this is a perfectly valid way to solve it. Anything I suggest is really a micro-optimisation or a readability fix.

The main thing here is you could reduce the level of nesting by just setting a fallback value for `name` is one is not set:

```golang
func foo(bar string) string {
  if bar == "" {
    bar = "fallback"
  }

  return fmt.Sprintf("bar is %s", bar")
}
```

Otherwise, think about what format verbs you're using in `Sprintf`. While %v does work, and in many cases is desireable, it is more of a "default" format. I prefer to use more specific verbs (%s for strings, %d for integers etc), in particular this is useful because if you pass the wrong type (say a string to a %d) the resulting string will show you that you made a mistake.